### PR TITLE
Patch ZipStreamer to fix directory downloads

### PR DESCRIPTION
**Pull request by nickvergessen**
Patches ZIpStreamer with https://github.com/McNetic/PHPZipStreamer/pull/39 to fix https://github.com/nextcloud/server/issues/2352

### CODE REVIEW

**Comment by MorrisJobke** [comment]
How to test this? The zip files work before and after this patch on my Windows 10 machine. With macOS they don't work (neither before nor after this patch).

**Comment by nickvergessen** [comment]
Ask people of https://github.com/nextcloud/server/issues/2352 for the setup configurations.
Or maybe @mathiasconradt can help with details

File: mcnetic/zipstreamer/src/ZipStreamer.php [hunk 1]
```diff
@@ -525,7 +525,7 @@ private function buildCentralDirectoryHeader($filePath, $timestamp, $gpFlags,
   private function buildEndOfCentralDirectoryRecord($cdRecLength) {
     if ($this->zip64) {
       $diskNumber = -1;
-      $cdRecCount = -1;
+      $cdRecCount =  min(sizeof($this->cdRec), 0xffff);;
```
Comment (nickvergessen) [inline mcnetic/zipstreamer/src/ZipStreamer.php]: same here

Comment (MorrisJobke) [reply mcnetic/zipstreamer/src/ZipStreamer.php]: Could you fix the extra space and the extra `;` and then this is good for me.

Comment (nickvergessen) [reply mcnetic/zipstreamer/src/ZipStreamer.php]: done in all 3 PRs

**Pull request merged**